File: frontend/src/types/feeRule.ts
export interface FeeRule {
  id: number;
  ruleNo: string;
  industryCode: string;
  industryName: string;
  payChannel?: string;
  payChannelDesc?: string;
  minAmount: number;
  maxAmount: number;
  feeRate: number;
  minFee: number;
  maxFee?: number;
  priority: number;
  status: number;
  statusDesc?: string;
  operatorName?: string;
  remark?: string;
  createdAt: string;
}

export interface FeeRuleQueryParams {
  current?: number;
  size?: number;
  industryCode?: string;
  payChannel?: string;
  status?: number;
}

export interface FeeRuleSaveRequest {
  id?: number;
  industryCode: string;
  industryName: string;
  payChannel?: string;
  minAmount: number;
  maxAmount: number;
  feeRate: number;
  minFee?: number;
  maxFee?: number;
  priority?: number;
  status?: number;
  remark?: string;
}

export interface FeeCalcRequest {
  industryCode: string;
  payChannel?: string;
  amount: number;
}

export interface FeeCalcResult {
  amount: number;
  feeAmount: number;
  feeRate: number;
  minFee?: number;
  maxFee?: number;
  ruleNo?: string;
  industryCode: string;
  industryName?: string;
  payChannel?: string;
  calcDetail: string;
}

export interface FeeRulePageResult {
  records: FeeRule[];
  total: number;
  current: number;
  size: number;
  pages: number;
}

export interface IndustryOption {
  code: string;
  name: string;
}

File: backend/merchant/src/main/java/com/payhub/merchant/entity/FeeRule.java
package com.payhub.merchant.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("fee_rule")
public class FeeRule implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId(type = IdType.AUTO)
    private Long id;

    private String ruleNo;

    private String industryCode;

    private String industryName;

    private String payChannel;

    private BigDecimal minAmount;

    private BigDecimal maxAmount;

    private BigDecimal feeRate;

    private BigDecimal minFee;

    private BigDecimal maxFee;

    private Integer priority;

    private Integer status;

    private String operatorId;

    private String operatorName;

    private String remark;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableLogic
    private Integer deleted;
}
